app/src/main/java/com/practicum/playlistmaker/media/presentation/FavoriteFragment.kt

```diff
@@ -2,19 +2,80 @@ package com.practicum.playlistmaker.media.presentation
 
 import android.os.Bundle
 import android.view.View
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.practicum.playlistmaker.main.presentation.TrackAdapter
+import com.practicum.playlistmaker.search.domain.Track
+import com.practicum.playlistmaker.search.presentation.SearchFragment
 import com.practicum.playlistmaker2.R
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class FavoriteFragment: Fragment(R.layout.fragment_favorites) {
+class FavoriteFragment : Fragment(R.layout.fragment_favorites) {
 
     private val viewModel: FavoritesViewModel by viewModel()
+    private var isClickAllowed = true
+    private var clickDebounceJob: Job? = null
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        val favoriteRecycler = view.findViewById<RecyclerView>(R.id.favorite_recycler)
+        val emptyListView = view.findViewById<LinearLayout>(R.id.emptyView)
+        val adapter = TrackAdapter()
+        adapter.itemClickListener = { _, track ->
+            if (clickDebounce()) openPlayer(track)
+        }
+        favoriteRecycler.adapter = adapter
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.favoriteTracksFlow.collect {
+                    if (it.isEmpty()) {
+                        favoriteRecycler.isVisible = false
+                        emptyListView.isVisible = true
+                    } else {
+                        adapter.trackList = it
+                        adapter.notifyDataSetChanged()
+                        favoriteRecycler.isVisible = true
+                        emptyListView.isVisible = false
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
        viewModel.favoriteTracksFlow
            .flowWithLifecycle(viewLifecycleOwner.lifecycle)
            .onEach {
                if (it.isEmpty()) {
                    favoriteRecycler.isVisible = false
                    emptyListView.isVisible = true
                } else {
                    adapter.trackList = it
                    adapter.notifyDataSetChanged()
                    favoriteRecycler.isVisible = true
                    emptyListView.isVisible = false
                }
            }
            .launchIn(viewLifecycleOwner.lifecycleScope)
```
То же самое, но более короткая запись и меньше вложенностей